### DISCUSSION TITLE
Searching for _Why_ Checks Failed

### DISCUSSION THREAD

**Discussion by matthugs** [community/community#31238] [Pull Requests]
I was reading https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-commit-status and I noticed that I can search for PRs which failed checks, but I can't filter to PRs which failed a _specific_ check (e.g. which failed for a specific reason, like a particular linter failing).

Use case/user story: I have heard from a coworker that a particular linter implemented across many repos in my Github org. produces bad output. I want to write a ticket to the sub-team of my organization which owns this linter, but I only have hearsay from my coworker to demonstrate that the linter produces bad/unuseful-to-the-developer output when the linter fails. I would like to search for PRs which failed checks (which I can do with a search like `org:openedx type:pr status:failure` [iiuc the semantics of that search, feel free to correct me if I'm wrong]), but filter specifically to PRs which failed checks _because_ of the linter I'm interested in.

This might be achieved by exposing a search field with semantics like `failing_check_name:` or something, e.g. my search might be refinable with like `org:openedx type:pr status:failure failing_check_name:check_reserved_keywords` to turn up things like (paste link here if I luck out & find one, lol, rip me) without PRs like https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/615/checks (which failed checks but not the check I'm interested in searching for.

Is there any way of achieving this with features as they currently exist? Or is this a feature request?

**Comment by airtower-luna** [comment]
I think the tricky thing here is to check which check involved which tool. If you have a standard within your organization that says the linter must be called in a separate job with a specific name it might work for a search within your org repositories, but in general people might call a linter in many different ways, so I think it'd be hard to automatically detect workflows using it, let alone if they failed because of the linter. :sweat_smile: 

On the other hand, maybe you don't need a completely reliable way. Something that gets you a sufficient sample would probably be enough. Maybe search commits with failed checks and the linter name occurring in workflow names, or something? If that's too much, you could analyze the PR checks via the API (e.g. find out which job failed, check if it's one involving the linter).

Though the first thing I'd do in your position is to ask that colleague if they have any examples. :wink:

**Reply by matthugs** [reply]
Yeah I'm imagining the search field indexing the name of the workflow/line-item-it-shows-up-as-a-check-under (lmk if unclear). B/c it won't be fully comprehensive 

I.e. if I was looking for stuff like https://github.com/openedx/credentials/pull/1741/checks I would search for `org:openedx type:pr status:failure failing_check_name:quality_and_translations_tests` since that's what shows up here:
<img width="903" alt="Screenshot 2022-08-29 at 11 36 00 AM" src="https://user-images.githubusercontent.com/6405097/187239020-d7b8c6ae-7e61-4479-89fd-e3e959d838b8.png">
